webdriverio/index.d.ts

```diff
@@ -352,36 +352,37 @@ declare namespace WebdriverIO {
         ): Client<P>;
         getHTML<P>(includeSelectorTag: boolean): Client<P>;
 
-        getLocation(selector?: string): Size;
-        getLocation(selector: string, axis: string): number;
-        getLocation(axis: string): number;
+        getLocation(selector?: string): Location;
+        getLocation(selector: string, axis: Axis): number;
+        getLocation(axis: Axis): number;
```

Comment:
Hi @DanielRosenwasser, thanks for taking the time to review this.
It also took me a while and some mistakes to understand this. It's explained in the api [main page](http://webdriver.io/api.html#Element-as-first-citizen) that all functions that receive a `selector` can be used curried if `element` is called first.

Probably there's a better way to model this, but it's out of the scope of this PR.
Should I invert the order of the declarations so `Axis` is before the more general `string` type?